Blocks/URL/test.cc

```diff
@@ -95,42 +101,58 @@ TEST(URLTest, CompositionTest) {
-  EXPECT_EQ("telnet://www.google.com:23/", URL("www.google.com", "telnet", "", 23).ComposeURL());
+  EXPECT_EQ("http://www.google.com/", URL("www.google.com", "").ComposeURL());
+  EXPECT_EQ("telnet://www.google.com/", URL("www.google.com", "telnet").ComposeURL());
+  EXPECT_EQ("telnet://www.google.com:23/", URL("www.google.com:23", "telnet").ComposeURL());
   // Keeps the scheme if it was explicitly specified, even for the port that maps to a different scheme.
-  EXPECT_EQ("foo://www.google.com:80/", URL("foo://www.google.com", "", "", 80).ComposeURL());
+  EXPECT_EQ("foo://www.google.com/", URL("foo://www.google.com").ComposeURL());
+  EXPECT_EQ("foo://www.google.com:80/", URL("foo://www.google.com:80").ComposeURL());
   // Maps port 80 into "http://".
+  EXPECT_EQ("http://www.google.com/", URL("www.google.com:80").ComposeURL());
+  EXPECT_EQ("http://www.google.com/", URL("www.google.com:80", "").ComposeURL());
+  EXPECT_EQ("http://www.google.com/", URL("www.google.com:80", "", "", 80).ComposeURL());
   EXPECT_EQ("http://www.google.com/", URL("www.google.com", "", "", 80).ComposeURL());
   // Maps port 443 into "https://".
+  EXPECT_EQ("https://www.google.com/", URL("www.google.com:443").ComposeURL());
+  EXPECT_EQ("https://www.google.com/", URL("www.google.com:443", "").ComposeURL());
+  EXPECT_EQ("https://www.google.com/", URL("www.google.com:443", "", "", 443).ComposeURL());
   EXPECT_EQ("https://www.google.com/", URL("www.google.com", "", "", 443).ComposeURL());
   // Assumes port 80 for "http://".
-  EXPECT_EQ("http://www.google.com:79/", URL("www.google.com", "http", "", 79).ComposeURL());
-  EXPECT_EQ("http://www.google.com/", URL("www.google.com", "http", "", 80).ComposeURL());
-  EXPECT_EQ("http://www.google.com:81/", URL("www.google.com", "http", "", 81).ComposeURL());
+  EXPECT_EQ("http://www.google.com:79/", URL("http://www.google.com:79").ComposeURL());
+  EXPECT_EQ("http://www.google.com/", URL("http://www.google.com:80").ComposeURL());
+  EXPECT_EQ("http://www.google.com:81/", URL("http://www.google.com:81").ComposeURL());
   // Assumes port 443 for "https://".
-  EXPECT_EQ("https://www.google.com:442/", URL("www.google.com", "https", "", 442).ComposeURL());
-  EXPECT_EQ("https://www.google.com/", URL("www.google.com", "https", "", 443).ComposeURL());
-  EXPECT_EQ("https://www.google.com:444/", URL("www.google.com", "https", "", 444).ComposeURL());
-  // Since there is no rule from "23" to "telnet", no scheme is specified.
-  EXPECT_EQ("www.google.com:23/", URL("www.google.com", "", "", 23).ComposeURL());
+  EXPECT_EQ("https://www.google.com:442/", URL("https://www.google.com:442").ComposeURL());
+  EXPECT_EQ("https://www.google.com/", URL("https://www.google.com:443").ComposeURL());
+  EXPECT_EQ("https://www.google.com:444/", URL("https://www.google.com:444").ComposeURL());
+  // Since there is no rule for port "23", no scheme is returned.
+  EXPECT_EQ("www.google.com:23/", URL("www.google.com:23").ComposeURL());
```

Comment:
In its previous version, this test was explicitly testing a behavior with URL ctor parameters dealing with redirects. I'd suggest to keep the separate tests for port-scheme mapping and tests for derived scheme.